streamer.c

```diff
@@ -167,32 +168,53 @@ void cLiveStreamer::Close(void)
 void cLiveStreamer::Action(void)
 {
   int ret;
-  sStreamPacket pkt;
-  memset(&pkt, 0, sizeof(sStreamPacket));
-  bool requestStreamChange = false;
+  sStreamPacket pkt_data;
+  sStreamPacket pkt_side_data; // Additional data
+  memset(&pkt_data, 0, sizeof(sStreamPacket));
+  memset(&pkt_side_data, 0, sizeof(sStreamPacket));
+  bool requestStreamChangeStdData = false;
+  bool requestStreamChangeExtData = false;
   cTimeMs last_info(1000);
   cTimeMs bufferStatsTimer(1000);
 
   while (Running())
   {
-    ret = m_Demuxer.Read(&pkt);
+    ret = m_Demuxer.Read(&pkt_data, &pkt_side_data);
     if (ret > 0)
     {
-      if (pkt.pmtChange)
+      if (pkt_data.pmtChange)
       {
-        requestStreamChange = true;
+        requestStreamChangeStdData = true;
+        requestStreamChangeExtData = true;
       }
-      if (pkt.data)
+
+      // Process normal data if present
+      if (pkt_data.data)
+      {
+        if (pkt_data.streamChange || requestStreamChangeStdData)
+          sendStreamChange();
+        requestStreamChangeStdData = false;
+        if (pkt_data.reftime)
+        {
+          sendRefTime(&pkt_data);
+          pkt_data.reftime = 0;
+        }
+        sendStreamPacket(&pkt_data);
+      }
+
+      // If some additional data is present inside the stream, it is written there (currently RDS inside MPEG2-Audio)
+      if (pkt_side_data.data)
       {
-        if (pkt.streamChange || requestStreamChange)
+        if (pkt_side_data.streamChange || requestStreamChangeExtData)
           sendStreamChange();
-        requestStreamChange = false;
-        if (pkt.reftime)
+        requestStreamChangeExtData = false;
+        if (pkt_side_data.reftime)
```

Comment:
For now it can be removed, let me check on stream after all new build are finished and I create a new pull request.